### PR TITLE
Add integration testing coverage for Problems API

### PR DESCRIPTION
**Pull request by hegyibalint**

### CODE REVIEW

File: subprojects/problems/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+                protected abstract Problems getProblems();
+
+                @TaskAction
+                void run() {
+                    Problem problem = problems.createProblemBuilder()
+                        .label("label")
+                        .undocumented()
+                        .noLocation()
+                        .type("type")
+                        .severity(Severity.ERROR)
```
Comment (donat) [inline subprojects/problems/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy:49]: severity is not mandatory

File: subprojects/problems/src/main/java/org/gradle/api/problems/Problems.java [hunk 2]
```diff
@@ -60,57 +55,29 @@ public abstract class Problems {
...
-    private static Problems problemsService = new NoOpProblems();
-
-    abstract public ProblemBuilderDefiningLabel createProblemBuilder();
-
-    abstract public void collectError(RuntimeException failure);
-
+    ProblemBuilderDefiningLabel createProblemBuilder();
 
-    abstract public void collectError(Problem problem);
+    void collectError(RuntimeException failure);
```
Comment (donat) [inline subprojects/problems/src/main/java/org/gradle/api/problems/Problems.java:66]: Please rebase to master. This has already been removed

File: subprojects/problems/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy (hunk 1 above)
Comment (hegyibalint) [reply subprojects/problems/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy:49]: Thanks, replaced with a separate test case 🎉

**Review by donat — APPROVED** [review]

**Comment by hegyibalint** [comment]
@bot-gradle test and merge

**Comment by hegyibalint** [comment]
@bot-gradle test and merge

**Pull request merged**